elastic/kibana
# [Security Solution][CTI] Update legacy CTI signals to latest ECS threat fields

## Summary

This PR Is split into two logical chunks, committed sequentially:
### Data Updates ([diff](https://github.com/rylnd/kibana/compare/ab43afab881da7e6935ebb569eb717691f8cadde...66d60e9a))
* Updated mappings for the latest [ECS CTI fields](https://github.com/elastic/ecs/pull/1504), aka `threat.enrichments`
    * Security Solution users will have their old signals index templates updated, and their current signals index will then roll over to one with `threat.enrichments` mappings
    * When RAC rules/alerts are enabled, the indices above will no longer be written to, but new, non-RAC signals will be queryable via `threat.enrichments` thanks to the index alias from RAC indices to older signals indices
* A "data migration" that converts legacy signals to the new format
    * This is required for users to "update" their old CTI signals, if they so choose. The expectation is that the UI will only support `threat.enrichments` moving forward, and so this will enable backwards compatibility.
    * This is implemented as a painless script executed as part of our Signals Migration API. Adding more and more code to this script is a terrible pattern that we should normally avoid, but in this case we mercifully have some mitigating factors that make it acceptable:
        1. 7.15 is the last minor release that will deal with the signals index; moving forward, all data will be written to RAC indices
        2. There are lots of guards to ensure we only run this on appropriate documents 
### App Updates ([diff](https://github.com/rylnd/kibana/compare/66d60e9a...2cdcd62))
  This involved updating application code to read/write from `threat.enrichments` instead of `threat.indicator`. This touched the following areas:
  * Threat Indicator Match Event Renderer
    * updated both fields queried and parsing logic
    * removed `dataset` field from event renderer as that is not part of ECS nor deemed relevant indicator info
  * Alerts Summary tab
    * Updated to parse CTI fields from both existing enrichments (`threat.enrichments`), legacy indicators (`threatintel.indicator`), and future indicators (`threat.indicator`)
  * Threat Intel Tab
    * Updated to parse CTI fields from both existing enrichments (`threat.enrichments`), legacy indicators (`threatintel.indicator`), and future indicators (`threat.indicator`)
    * Fields displayed now begin with `indicator` or `matched`, corresponding to ECS `threat.enrichments`
  * Prebuilt timeline template
    * Updated query analogous fields from `threat.enrichments.matched` instead of `threat.indicator.matched`
    

## Screenshots
### new row renderer (sans dataset field)
<kbd>
<img width="1227" alt="Screen Shot 2021-08-12 at 9 32 00 PM" src="https://user-images.githubusercontent.com/657252/129296837-1b4f3f98-9c9f-4fcd-a7f4-e32228fbe694.png">
</kbd>

### Existing enrichments in Threat Intel tab
<kbd>
<img width="755" alt="Screen Shot 2021-08-12 at 9 33 58 PM" src="https://user-images.githubusercontent.com/657252/129296917-37581acd-2808-4f65-8124-735895aff45e.png">
</kbd>
### 7.14 filebeat indicators in Threat intel tab
<kbd>
<img width="755" alt="Screen Shot 2021-08-12 at 9 36 04 PM" src="https://user-images.githubusercontent.com/657252/129297073-db891b32-1324-4115-aa26-e58d25159d34.png">
</kbd>

## Review steps
1. Checkout `7.15` branch; create an indicator match rule and generate alerts
2. Checkout this branch _without disabling the rule_.
    * This will cause the rule to generate `threat.enrichments` alerts in the existing index
3. Visit the detections page to initiate a template update/rollover
    * `threat.enrichments` alerts will now have the proper mappings
4. Verify that "old" alerts are still visible and do not cause errors
    * The "Show only threat indicator alerts" filter should still include them
    * Threat Row renderer should not work
    * Threat Intel tab should show 0 threats
5. Verify that "new" alerts are fully functional
    * The "Show only threat indicator alerts" filter should include them
    * Threat Row renderer should work
    * Threat Intel tab should show N threats
6. Use the [Migration API]() to migrate old signals in 4. to the new format
7. Verify the newly-migrated signals now behave as the "new" alerts in 5.

### Checklist




### For maintainers
